doc/sphinx/language/core/basic.rst

```diff
@@ -146,6 +148,29 @@ Numbers
   The :cmd:`Number Notation` mechanism offers the user
   a way to define custom parsers and printers for :n:`@number`.
 
+  By default, numbers are interpreted as :n:`nat`\s, which is a unary
+  representation.  For example, :n:`3` is represented as `S (S (S O))`.  While
+  this is a convenient representation for doing proofs, computing with large
+  :n:`nat`\s can lead to stack overflows or running out of memory.  You can
+  explicitly specify a different interpretation to avoid this problem.  For
+  example, :n:`1000000%N` efficiently represents that natural number.  See
+  :ref:`Scopes` and :n:`@term_scope`.  You can also change the default
+  interpretation using the :cmd:`Number Notation` command.
```

Comment:
This sentence is already at the end of the previous paragraph. Moreover, the NUmber Notation command rather targets a few specific users (library authors (honestly, even me as coauthor of the command, I probably don't use it more than once a year ;) )). The casual user is probably more interested in the fact that multiple libraries offer her a wide range of options. Maybe something along the lines: "Multiple libraries offer numbers with different ease of proof/efficiency of computation trade-offs, ranging from the unary nat in the stdlib (https://coq.gitlab.io/-/coq/-/jobs/3977290672/artifacts/_build/default/doc/refman-html/language/coq-library.html#basic-arithmetic) to arbitrary precision machine integers in the bignums library (https://github.com/coq/bignums/)".

Reply:
Yeah, doesn't need to be repeated, I will adjust.

How does `Z` differ to the user from bignums?

I'm reluctant to start enumerating alternative libraries here.  There should be another section elsewhere that mentions all the alternatives and how to choose between them, which we could link to if it existed.  This part of the chapter primarily introduces syntax.  Getting into specific libraries is too much detail.  I don't want to expand the scope of the PR to create such a section right now.

Reply:
> How does `Z` differ to the user from bignums?

`Z` are binary integers, that is one constructor (a few machine words) for each bit of the number where bignums offers a machine integer encoding, that is a constructor for each packet of 63 bits. The second is then much more efficient computationally speaking.

> I'm reluctant to start enumerating alternative libraries here. There should be another section elsewhere that mentions all the alternatives and how to choose between them, which we could link to if it existed. This part of the chapter primarily introduces syntax. Getting into specific libraries is too much detail. I don't want to expand the scope of the PR to create such a section right now.

I agree a full list would require work, I was just expecting we could add a few, definitely non exhaustive, examples to make clear to the reader that numbers are a library thing and that multiple options are available around (which is probably something not that common among usual programming languages, hence the need to pinpoint it IMHO).